packages/hs-react-ui/src/components/Table/Table.stories.tsx

```diff
@@ -182,3 +223,136 @@ storiesOf('Table', module).add(
+      });
+      setRows(newRows);
+    };
+
+    const SelectAllCell = () => {
+      let totalSelected = 0;
+      let totalCheckboxesAccumulator = 0;
+      rows.forEach(groupRows => {
+        groupRows.forEach((row: columnTypes) => {
+          if (groupRows.isGroupLabel) return;
+          if (row.selected) totalSelected++;
+          totalCheckboxesAccumulator++;
+        });
+      });
+      const allChecked = totalSelected === totalCheckboxesAccumulator;
+
+      // TODO: don't use pointer-events to control if a column is sortable - it should be checked
+      // within the sorting listeners so that `sortable` doesn't need to be passed here just to give
+      // it pointer-events
+      return (
+        <Table.HeaderCell sortable>
+          <Checkbox
+            checkboxType={allChecked ? CheckboxTypes.check : CheckboxTypes.neutral}
+            checked={Boolean(totalSelected)}
+            onClick={selectAll}
+          />
+        </Table.HeaderCell>
+      );
+    };
+
+    const SelectionCell = ({
+      index,
+      selected,
+      reachedMinWidth,
+      groupIndex,
+    }: {
+      index: number;
+      selected: boolean;
+      reachedMinWidth?: boolean;
+      groupIndex?: number;
+    }) => (
+      <Table.Cell>
+        <Checkbox
+          onClick={() => onSelect(index, groupIndex, selected)}
+          checkboxType={CheckboxTypes.check}
+          checked={selected}
+        >
+          {reachedMinWidth ? 'Select for download' : ''}
+        </Checkbox>
+      </Table.Cell>
+    );
+
+    const EmptyCell = () => <Table.Cell />;
+
+    const NotesCell = ({ notes }: { notes: string }) => (
+      <Table.Cell>
+        <NoteField onChange={() => console.log()} rows={3} value={notes} />
+      </Table.Cell>
+    );
+
+    const sampleColumns: { [index: string]: any } = {
+      selection: {
+        name: '',
+        headerCellComponent: SelectAllCell,
+        width: text('Selection width', '2rem'),
+        cellComponent: SelectionCell,
+        sortable: false,
+        groupCellComponent: EmptyCell,
+      },
+      name: {
+        name: 'Name',
+        width: text('Name width', '1fr'),
+      },
+      title: {
+        name: 'Title',
+        width: text('Title width', '1fr'),
+      },
+      address: {
+        name: 'Address',
+        width: text('Address width', '1fr'),
+      },
+      notes: {
+        name: 'Notes',
+        width: text('Notes width', '12rem'),
+        cellComponent: NotesCell,
+        minTableWidth: 800,
+        sortFunction: (a: string, b: string) => (a.length > b.length ? -1 : 1),
+        groupCellComponent: EmptyCell,
+      },
+    };
+    const position = select('groupLabelPosition', groupLabelPositionOptions, 'above');
+    return (
+      <Table
+        columns={sampleColumns}
+        data={rows}
+        sortGroups={boolean('sortGroups', false)}
+        groupHeaderPosition={position}
+        isCollapsable={boolean('isCollapsable', false)}
+        minWidthBreakpoint={0}
```

Comment:
Do groups not work with the responsive version of the table? We should probably default to 0 or -1 or something so we don't have to pass this just to avoid the mobile version.

but groups should probably support responsive modes as well

Reply:
You know what, I completely forgot about this after I got everything working. I was having issues with the groups when all of the groups were collapsed. I'll check more into this